lib/daru/io/importers/rails.rb

```diff
@@ -0,0 +1,150 @@
+                end
+              elsif header_line?(request_data)
+                if @current_request
+                  case options[:parse_strategy]
+                  when 'assume-correct'
+                    handle_request(@current_request, &block)
+                    @current_request = @file_format.request(request_data)
+                  when 'cautious'
+                    @skipped_lines += 1
+                    warn(:unclosed_request, "Encountered header line (#{request_data[:line_definition].name.inspect}), but previous request was not closed!")
+                    @current_request = nil # remove all data that was parsed, skip next request as well.
+                  end
+                elsif footer_line?(request_data)
+                  handle_request(@file_format.request(request_data), &block)
+                else
+                  @current_request = @file_format.request(request_data)
+                end
+              else
+                if @current_request
+                  @current_request << request_data
+                  if footer_line?(request_data)
+                    handle_request(@current_request, &block) # yield @current_request
+                    @current_request = nil
+                  end
+                else
+                  @skipped_lines += 1
+                  warn(:no_current_request, "Parseable line (#{request_data[:line_definition].name.inspect}) found outside of a request!")
+                end
+              end
+              a = @file_format.request(request_data).attributes
+              @parsed_hash << a if not a.nil?
+            end
+          end
+        end
+
+        def read(path)
+          @path = path
+          @file_data = parse(@path)
+          self
+        end
+
+        def call()
+          ind = [:method, :path, :ip, :timestamp, :line_type, :lineno, :source,
+                 :controller, :action, :format, :params, :rendered_file,
+                 :partial_duration, :status, :duration, :view, :db]
+          data = Daru::DataFrame.new({},index: ind).transpose
+          @file_data.each do |hash|
+            row = []
+            ind.each do |attr|
+              row << hash[attr]
+            end
+            data.add_row row
+          end
+          data
+        end
+
+        def parse(path)
+          subject = RequestLogAnalyzer::FileFormat.load(:rails3)
+          log_parser = RequestLogAnalyzer::Source::LogParser.new(subject)
+          raw_list = log_parser.parse_file(path)
+          parsed_list = []
+          raw_list.each do |hash|
+            parsed_list << hash if hash.key? :method
+          end
+          for i in (0...parsed_list.size - 1)
+            j = raw_list.index(parsed_list[i+1])
+            k = raw_list.index(parsed_list[i]) + 1
+            for l in (k...j)
+              parsed_list[i].merge!(raw_list[l])
+            end
+          end
+          parsed_list
+        end
+      end
+    end
+  end
```

Comment:
In fact, here you are copy-pasting and slightly changing code from other gem. It is totally unreliable: if the gem changes a slightest bit, you'll never follow this with your code. Why do you need this rewriting?

Reply:
I will try to come up with a single method instead of changing the code.